wrapper/src/main/java/software/amazon/jdbc/util/WrapperUtils.java

```diff
@@ -449,14 +476,7 @@ public static <T> T createInstance(
         return resultClass.cast(classToInstantiate.newInstance());
       }
 
-      Class<?>[] argClasses = constructorArgClasses;
-      if (argClasses == null) {
-        argClasses = new Class<?>[constructorArgs.length];
-        for (int i = 0; i < constructorArgs.length; i++) {
-          argClasses[i] = constructorArgs[i].getClass();
-        }
-      }
-      final Constructor<?> constructor = classToInstantiate.getConstructor(argClasses);
+      final Constructor<?> constructor = classToInstantiate.getConstructor(constructorArgClasses);
```

Comment:
I'd be careful with this change. There was a bug. Please double check:

https://github.com/aws/aws-advanced-jdbc-wrapper/commit/ca0b8fdf2f42bbea1e315c162cfb679b8430af66#diff-d0ef1b1788499382bf76f53bc7fc3cbf0fed1a23fc0896c10d2d7afa34202d0bR284-R289